### ISSUE TITLE
[a11y]: DataTable [dynamic] - SR table Row announcement

### ISSUE THREAD

**Issue by andreancardona** [carbon-design-system/carbon#14261]
### Package

@carbon/react

### Browser

FF & Safari & iPhone

### Package version

@carbon/react@1.32.0

### Automated testing tool and ruleset

Manual QA

### Assistive technology

JAWS & VO

### Description
SR table Row announcement

SR announces the number of Rows incorrectly when the table is focused

When the table is focused, SR announces the number of columns in the table incorrectly. It appears that the SR counts the visually hidden rows as well.

### Recommended Fix
Screen reader is expected to identify the correct number of Rows in the table

If the table structure is not conveyed properly, it’s difficult for screen reader users to navigate through the table content and correlate data. 

### WCAG 2.1 Violation

1.3.1 Info and Relationships

### Reproduction/example
Record `Number 64` in the below table:
https://airtable.com/shrFsaeHhImOkuVNR

https://react.carbondesignsystem.com/?path=/story/components-datatable-dynamic--default

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by francinelucca** [comment]
Announcement of row numbers: press “t” to go to table. Reads out 8 columns and 13 rows when there are less than 13 rows. This is probably due to the expandable content that is hidden being recognized as a row
